packages/theme-language-server-common/src/docset/MarkdownRenderer.ts

```diff
@@ -67,6 +68,22 @@ function htmlEntryBody(entry: HtmlEntry, parentEntry?: HtmlEntry) {
     .join(HORIZONTAL_SEPARATOR);
 }
 
+function syntax(entry: DocsetEntry | FilterEntry | TagEntry) {
+  if (!('syntax' in entry) || !entry.syntax) {
+    return undefined;
+  }
+
+  // TagEntry entries already have liquid tags as a part of the syntax
+  // explanation so we can return them directly.
+  if (entry.syntax.startsWith('{%')) {
+    return `\`\`\`liquid\n${entry.syntax}\n\`\`\``;
+  }
+
+  // Wrap the syntax in liquid tags to ensure we get proper syntax highlighting
+  // if it's available.
+  return `\`\`\`liquid\n{{ ${entry.syntax} }}\n\`\`\``;
+}
```

Comment:
Given that we have fallback tag.json and filter.json files in packages/theme-liquid-docs-updater/data. It feels like we could benefit from having a test in there that validates the assumption.

Reply:
We chatted about this IRL. Going to skip this for now but it's good to keep in mind anytime we make assumptions about the data generated by theme docs. Fortunately for this usage the worst case scenario is syntax highlighting not quite working.